File: backend/components/metiers/threed/security/docs/samples/rbac/samples/sample_rbac.py
# sample_rbac.py
# Exemple d’intégration avancée pour la gestion des rôles

from helpers_rbac import assign_role, check_permission

user = {'id': 'u123', 'name': 'Alice', 'permissions': ['read']}
user = assign_role(user, 'admin')
print('Roles:', user['roles'])
print('Has read:', check_permission(user, 'read'))
print('Has write:', check_permission(user, 'write'))

File: backend/components/metiers/threed/security/docs/samples/rbac/samples/sample_rbac.js
// sample_rbac.js
// Exemple d’intégration avancée pour la gestion des rôles

const { assignRole, checkPermission } = require('./helpers_rbac');

let user = { id: 'u123', name: 'Alice', permissions: ['read'] };
user = assignRole(user, 'admin');
console.log('Roles:', user.roles);
console.log('Has read:', checkPermission(user, 'read'));
console.log('Has write:', checkPermission(user, 'write'));
